### DISCUSSION TITLE
Distinct job for each schedule

### DISCUSSION THREAD

**Discussion by mhitza** [community/community#25662] [Actions]
<p>I have a backup.yml workflow which I use for automated database backup. I’ve started with a workflow that schedules hourly, however, I want to include in a single workflow multiple backup jobs. Hourly, daily, weekly, etc.</p>
<p>Is there a way for me to conditionally skip a job in the workflow if the workflow was not triggered by the desired schedule? For example:</p>
<pre><code>jobs:
  hourly:
    if: invoked by an hourly trigger
  daily:
    if: invoked by a daily trigger
</code></pre>
<p>  </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mhitza">@mhitza</a> ,</p>
<p>Thank you for being here! If you’d like a single workflow to include multiple schedule jobs, you can add multiple ‘corn’ expression for schedule event, sample as below, you can use ‘cron: ‘*/60 * * * *’’ for hourly…etc.(doc <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule" rel="nofollow noopener">here</a>)</p>
<pre><code>on:
  schedule:
    - cron: '*/5 * * * *' # every 5 mins
    - cron: '*/7 * * * *' # every 7 mins
</code></pre>
<p>For different schedule, the github has different context, for intance, 7mins schedule:</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181092412-a4ac42d6-4dca-4823-a4be-c1529f26ed96.png" alt="schedulecheck.png" data-base62-sha1="4p6nHQCpqgy6khlBWJDsfZpzLhE" width="481" height="387"></p>
<p>You can use if expression to conditionally skip the other jobs then.</p>
<pre><code>on:
  schedule:
    - cron: '*/5 * * * *'
    - cron: '*/7 * * * *'
jobs:
  job1:
    if: contains(github.event.schedule, '*/7 * * * *')
    runs-on: [ubuntu-latest]
    steps:
      - name: response when schedule 7 mins
        run: |
          echo this is 7 minutes schedule
  job2:
    if: contains(github.event.schedule, '*/5 * * * *')
    runs-on: [ubuntu-latest]
    steps:
      - name: response when schedule 5 mins
        run: |
          echo this is 5 minutes schedule
</code></pre>
<p>Hope it helps.</p>

**Comment by mrchief** [comment]
<p>Adding this to the official docs would be mighty helpful!</p>

**Comment by dhimmel** [comment]
<p>Noting <a href="https://github.community/t/multiple-schedules-for-workflow/154811">154811</a> and <a href="https://github.community/t/how-to-schedule-and-distinguish-multiple-cron-in-the-same-yml-file/167893">167893</a> which are similar questions.</p>
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> thanks for sharing a solution that allows:</p>
<ol>
<li>multiple cron schedules</li>
<li>conditional behavior based on which schedule triggered the workflow</li>
</ol>
<p>The use case for 2 (conditional behavior) would be assisted with the ability to assign a unique ID to each schedule.</p>
<pre><code class="lang-yaml">on:
  schedule:
    - name: job_x
      cron: '1 0 1 * *' # once a month
    - name: job_y
      cron: '1 0 1 * *' # also once a month, schedule value not unique
</code></pre>
<p>Then we could add conditions based on <code>github.event.schedule.name</code> (or whatever this gets called), which would be both more readable and allow for conditional behavior for multiple redundant schedules .</p>

**Reply by dimich99** [reply]
github.event.schedule.name - is this implemented?

**Reply by dhimmel** [reply]
No `github.event.schedule.name` was an enhancement request.

Currently, the `schedule` context includes a field like

```
github.event.schedule = "51 17 * * 3"
```

as seen in [this context log](https://github.com/dhimmel/dump-actions-context/blob/61784d02249c49ca1ba507fdb0b14d5babe50e5e/contexts/schedule.gron#L124) that tracks the current fields in GitHub action contexts.

**Comment by wazawoo** [comment]
<p>Any reasoning on using <code>contains</code> vs <code>==</code>?</p>
